controllers/operatorconfigmap_controller.go

```diff
@@ -123,204 +124,196 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (c *OperatorConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	var err error
 	c.ctx = ctx
-	c.log = log.FromContext(ctx, "ClusterVersion", req)
-	c.log.Info("Reconciling ClusterVersion")
+	c.log = log.FromContext(ctx, "OperatorConfigMap", req)
+	c.log.Info("Reconciling OperatorConfigMap")
 
-	if err := c.ensureConsolePlugin(); err != nil {
-		c.log.Error(err, "unable to deploy client console")
-		return ctrl.Result{}, err
-	}
-
-	instance := configv1.ClusterVersion{}
-	if err = c.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
-		return ctrl.Result{}, err
-	}
+	// deletion phase
+	if !c.OperatorConfigMap.GetDeletionTimestamp().IsZero() {
+		if err := c.deletionPhase(); err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
I don't think this block will ever trigger, did you observe this in your testing? The reconciler is storing the configmap once at the program start and not refreshing after that so on the resource that's stored in the reconciler the deletion time stamp will never be set.